### DISCUSSION TITLE
What's the flow for generating internal Zettlekasten style links

### DISCUSSION THREAD

**Discussion by hajush** [Zettlr/Zettlr#3139] [Q&A] [answered]
I'm really loving Zettlr. I'm not great at a slipbox, but the ideas seem so valid. I get the value of note sequences, which in a physical slipbox would just be inserted in order between slips. The Zettlr electronic equivalent is to simple link between notes using the id (the file name). I'm happy to do that. I'm just not quite finding a flow in it and I think I'm missing something obvious that I'm not getting from the user documents.

I type the `[[`. But how do I quickly get the id's of other notes? I can type in text there and search. But they pop into the left search pane without any obvious way to copy the ids (or file names). What clue would make me less clue-less here?

**Comment by kyaso** [comment, marked as answer]
A pop-up should appear near your cursor when you type `[[`. There you can select the file you want to link to. If you keep typing, the list will even further narrow down to only the files with (partially) matching names.

**Reply by hajush** [reply]
Thanks for the solution. The popup did actually did *not* appear when I had this frustration. I started finally experimenting with slipbox note sequences, read the tutorial again, and typed [[ and nothing. After asking my question, I continued with my note and got stuck with a bibliography reference `[@` and no popup appeared. It had worked before. My Zettlr must have gotten in a broken state somehow. When I restarted it was healthy for bibliography references. And the popup with `[[` also showed up just now.

**Comment by nathanlesage** [comment]
My two cents after having played with Zettelkasten style systems for the past five years: IDs are not really necessary. I've completely thrown all of these out. Filenames are much more useful. After this, kyaso did answer quite right: It's a quick and dirty search in the autocomplete dropdown. Just complete the suggestion with Enter or Tab and you're done!

**Reply by hajush** [reply]
Thanks @nathanlesage ! I appreciate the experience report. I decided to continue with these ID's for now, but I suspect you are right.

**Reply by nathanlesage** [reply]
The good thing at the system Zettlr employs is: You can switch from IDs to filenames at any time without any problems, because the old IDs just continue to work!

**Reply by kyaso** [reply]
When I started using Zettlr almost two years ago, the unique IDs caught my attention. I still use IDs over filenames, so I don't have to worry about broken links when I decide to rename a file.

**Reply by hajush** [reply]
Maybe I should post another question about this, but it is related to the flow for linking. Is there a way to restrict reference search within a workspace. I have logging workspaces that I work with during the day but are not for linking, and they clutter up my popup queue so relevant slipbox workspace entries don't even show when I type '[['. I know I could fix this by closing the workspace (which would slow me down quite a bit), but maybe I'm missing something again.

**Reply by nathanlesage** [reply]
Feel free to always suggest improvements

**Reply by gour** [reply]
Hello,

> When I started using Zettlr almost two years ago, the unique IDs caught my attention. I still use IDs over filenames, so I don't have to worry about broken links when I decide to rename a file.

I'm just starting with Zettlr which I find is a terrific piece of software and wonder how can one rename a file/note and **not** break links between them? Is there some mechanism to update all the links when renaming files?

**Reply by hajush** [reply]
It doesn't look like there is a way to do this with Zettlr, which seems like a useful feature request. This feature should be easy to implement. The "find in file" has a replace option. They just need to add it to "find in directory".

But you could do it with other text editors (like Sublime) as a directory wide find and replace. Or if you want to dive into the command line and regular expressions (regex), you could do it with sed/awk etc. Regex isn't much fun though - easy to get wrong and that can be dangerous especially if you are doing edits.
